src/mapper.rs

```diff
@@ -0,0 +1,170 @@
+    members: HashMap<&'s str, Vec<MemberMapping<'s>>>,
+}
+
+/// A Proguard Remapper.
+///
+/// This can remap frames one at a time, or the complete raw stacktrace.
+#[derive(Debug)]
+pub struct Mapper<'s> {
+    classes: HashMap<&'s str, ClassMapping<'s>>,
+}
+
+impl<'s> Mapper<'s> {
+    /// Create a new Proguard Remapper.
+    pub fn new(mapping: &'s str) -> Self {
+        let mut classes = HashMap::new();
+        let mut class = ClassMapping {
+            original: "",
+            obfuscated: "",
+            members: HashMap::new(),
+        };
+
+        for record in mapping.lines().filter_map(parse_mapping_line) {
+            match record {
+                MappingRecord::Header { .. } => {}
+                MappingRecord::Class {
+                    original,
+                    obfuscated,
+                } => {
+                    if !class.original.is_empty() {
+                        classes.insert(class.obfuscated, class);
+                    }
+                    class = ClassMapping {
+                        original,
+                        obfuscated,
+                        members: HashMap::new(),
+                    }
+                }
+                MappingRecord::Member {
+                    original,
+                    obfuscated,
+                    original_class,
+                    line_mapping,
+                    ..
+                } => {
+                    let (startline, endline, original_startline, original_endline) = line_mapping
+                        .map_or((0, 0, 0, None), |line_mapping| {
+                            (
+                                line_mapping.startline,
+                                line_mapping.endline,
+                                line_mapping.original_startline.unwrap_or(0),
+                                line_mapping.original_endline,
+                            )
+                        });
+                    let members = class.members.entry(obfuscated).or_insert_with(|| vec![]);
+                    members.push(MemberMapping {
+                        startline,
+                        endline,
+                        original_class,
+                        original,
+                        original_startline,
+                        original_endline,
+                    });
+                }
+            }
+        }
+        if !class.original.is_empty() {
+            classes.insert(class.obfuscated, class);
+        }
+
+        Self { classes }
+    }
+
+    /// Remaps a single Stackframe.
+    ///
+    /// Returns one or more [`StackFrame`]s, based on the information in
+    /// the proguard mapping. This can return more than one frame in the case
+    /// of inlined functions. In that case, frames are sorted top to bottom.
+    ///
+    /// [`StackFrame`]: struct.StackFrame.html
+    pub fn remap_frame(
+        &'s self,
+        frame: &StackFrame<'s>,
+    ) -> impl Iterator<Item = StackFrame<'s>> + 's {
```

Comment:
As discussed offline, it might be better to have separate methods for mapping a class and mapping a method, since both have different inputs and different outputs.